src/validation/entities/schemas/signature.schema.ts

```diff
@@ -0,0 +1,10 @@
+import { HexSchema } from '@/validation/entities/schemas/hex.schema';
+
+function isSignature(value: `0x${string}`): boolean {
+  // We accept proposals of singular or concatenated signatures
+  return (value.length - 2) % 130 === 0;
+}
+
+export const SignatureSchema = HexSchema.refine(isSignature, {
+  message: 'Invalid signature',
+});
```

Comment:
By validating signature length, we prevent unnecessary verification from happening. This also has the added benefit that we get more descriptive error messages.